ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/git/GitServiceClientImpl.java

```diff
@@ -445,69 +419,60 @@ public void push(DevMachine devMachine,
     }
 
     @Override
-    public Promise<List<Remote>> remoteList(DevMachine devMachine, Path project, String remote, boolean verbose) {
-        RemoteListRequest remoteListRequest = dtoFactory.createDto(RemoteListRequest.class).withVerbose(verbose);
-        if (remote != null) {
-            remoteListRequest.setRemote(remote);
-        }
-        String url = devMachine.getWsAgentBaseUrl() + REMOTE_LIST + "?projectPath=" + project;
-        return asyncRequestFactory.createPostRequest(url, remoteListRequest)
+    public Promise<List<Remote>> remoteList(DevMachine devMachine, Path project, String remoteName, boolean verbose) {
+        String params = "?projectPath=" + project.toString() + (remoteName != null ? "&remoteName=" + remoteName : "") +
+                        "&verbose=" + String.valueOf(verbose);
+        String url = appContext.getDevMachine().getWsAgentBaseUrl() + REMOTE + params;
+        return asyncRequestFactory.createGetRequest(url)
                                   .loader(loader)
                                   .send(dtoUnmarshallerFactory.newListUnmarshaller(Remote.class));
     }
 
-    /** {@inheritDoc} */
     @Override
     public void branchList(DevMachine devMachine,
                            ProjectConfig project,
-                           @Nullable String remoteMode,
+                           BranchListMode listMode,
                            AsyncRequestCallback<List<Branch>> callback) {
-        BranchListRequest branchListRequest = dtoFactory.createDto(BranchListRequest.class).withListMode(remoteMode);
-        String url =appContext.getDevMachine().getWsAgentBaseUrl() + BRANCH_LIST + "?projectPath=" + project.getPath();
-        asyncRequestFactory.createPostRequest(url, branchListRequest).send(callback);
+        String url = appContext.getDevMachine().getWsAgentBaseUrl() + BRANCH + "?projectPath=" + project.getPath() + "&listMode=" +
+                     listMode;
+        asyncRequestFactory.createGetRequest(url).send(callback);
     }
 
     @Override
-    public Promise<List<Branch>> branchList(DevMachine devMachine, Path project, String mode) {
-        BranchListRequest branchListRequest = dtoFactory.createDto(BranchListRequest.class).withListMode(mode);
-        String url = devMachine.getWsAgentBaseUrl() + BRANCH_LIST + "?projectPath=" + project;
-        return asyncRequestFactory.createPostRequest(url, branchListRequest).send(dtoUnmarshallerFactory.newListUnmarshaller(Branch.class));
+    public Promise<List<Branch>> branchList(DevMachine devMachine, Path project, BranchListMode listMode) {
+        String url = appContext.getDevMachine().getWsAgentBaseUrl() + BRANCH + "?projectPath=" + project.toString() + "&listMode=" +
+                     listMode;
+        return asyncRequestFactory.createGetRequest(url).send(dtoUnmarshallerFactory.newListUnmarshaller(Branch.class));
     }
 
     @Override
     public Promise<Status> status(DevMachine devMachine, ProjectConfig project) {
```

Comment:
I think that `getStatus` and `status` methods can reuse common code and should be named the same